plugins/OnlineSales.Plugin.Sms/Services/AmazonSnsService.cs

```diff
@@ -17,13 +21,34 @@ public AmazonSnsGatewayService(AmazonSnsConfig amazonSns)
         this.amazonSns = amazonSns;
     }
 
-    public Task SendAsync(string recipient, string message)
+    public async Task SendAsync(string recipient, string message)
     {
-        Log.Information("Sms message sent to {0} via AmazonSns gateway: {1}", recipient, message);
+        var accessKey = amazonSns.AccessKeyId;
+        var secretKey = amazonSns.SecretAccessKey;
+        RegionEndpoint region = RegionEndpoint.GetBySystemName(amazonSns.DefaultRegion);
 
-        // TODO: Implement real SMS handling via AWS
+        var client = new AmazonSimpleNotificationServiceClient(accessKey, secretKey, region);
+        var messageAttributes = new Dictionary<string, MessageAttributeValue>();
+        var smsType = new MessageAttributeValue
+        {
+            DataType = "String",
+            StringValue = "Transactional",
+        };
 
-        return Task.Delay(0);
+        messageAttributes.Add("AWS.SNS.SMS.SMSType", smsType);
+
+        PublishRequest request = new PublishRequest
+        {
+            Message = message,
+            PhoneNumber = recipient,
+            MessageAttributes = messageAttributes,
+        };
+
+        var response = await client.PublishAsync(request);
```

Comment:
Where is error handling? What happens if AWS returns an error? Please throw exception (see Smsc or Getshoutout implementation as a reference).